File: VECTOAux/VectoAuxiliaries/IAuxiliaryConfig.vb
' Copyright 2017 European Union.
' Licensed under the EUPL (the 'Licence');
'
' * You may not use this work except in compliance with the Licence.
' * You may obtain a copy of the Licence at: http://ec.europa.eu/idabc/eupl
' * Unless required by applicable law or agreed to in writing,
'   software distributed under the Licence is distributed on an "AS IS" basis,
'   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
'
' See the LICENSE.txt for the specific language governing permissions and limitations.

Imports VectoAuxiliaries.Electrics
Imports VectoAuxiliaries.Pneumatics
Imports VectoAuxiliaries.Hvac
Imports VectoAuxiliaries.DownstreamModules
Imports System.IO

Imports System.Windows.Forms
Imports Newtonsoft.Json


Public Interface IAuxiliaryConfig
	'Vecto
	Property VectoInputs As IVectoInputs

	'Electrical
	property ElectricalUserInputsConfig As IElectricsUserInputsConfig


	'Pneumatics
	Property PneumaticUserInputsConfig As IPneumaticUserInputsConfig
	Property PneumaticAuxillariesConfig As IPneumaticsAuxilliariesConfig

	'Hvac
	Property HvacUserInputsConfig As IHVACUserInputsConfig

	Function ConfigValuesAreTheSameAs(other As AuxiliaryConfig) As Boolean


	'Persistance Functions
	Function Save(filePath As String) As Boolean
	Function Load(filePath As String) As Boolean
End Interface

File: VECTOAux/BusAuxiliaries/IAuxiliaryConfig.cs
// Copyright 2017 European Union.
// Licensed under the EUPL (the 'Licence');
// 
// * You may not use this work except in compliance with the Licence.
// * You may obtain a copy of the Licence at: http://ec.europa.eu/idabc/eupl
// * Unless required by applicable law or agreed to in writing,
// software distributed under the Licence is distributed on an "AS IS" basis,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// 
// See the LICENSE.txt for the specific language governing permissions and limitations.

using TUGraz.VectoCore.BusAuxiliaries.Interfaces;
using TUGraz.VectoCore.BusAuxiliaries.Interfaces.DownstreamModules.Electrics;
using TUGraz.VectoCore.BusAuxiliaries.Interfaces.DownstreamModules.HVAC;
using TUGraz.VectoCore.BusAuxiliaries.Interfaces.DownstreamModules.PneumaticSystem;

namespace TUGraz.VectoCore.BusAuxiliaries {
	public interface IAuxiliaryConfig
	{
		// Vecto
		IVectoInputs VectoInputs { get; set; }

		// Electrical
		IElectricsUserInputsConfig ElectricalUserInputsConfig { get; set; }


		// Pneumatics
		IPneumaticUserInputsConfig PneumaticUserInputsConfig { get; set; }
		IPneumaticsAuxilliariesConfig PneumaticAuxillariesConfig { get; set; }

		// Hvac
		IHVACUserInputsConfig HvacUserInputsConfig { get; set; }

		bool ConfigValuesAreTheSameAs(AuxiliaryConfig other);


		// Persistance Functions
		bool Save(string filePath);
		bool Load(string filePath);
	}
}
